lib/chef/exceptions.rb

```diff
@@ -212,7 +212,11 @@ class ChildConvergeError < RuntimeError; end
 
     class NoProviderAvailable < RuntimeError; end
 
-    class DeprecatedFeatureError < RuntimeError; end
+    class DeprecatedFeatureError < RuntimeError;
+      def initalize
```

Comment:
Does this ignore the message passed in? I'd like to keep that and maybe add the text below as a parenthetical, e.g. `Chef::Log.deprecation("this thing is bad, do xyz(object) instead")` would give you `"this thing is bad, do 'xyz(object)' instead (raising deprecation error because treat_deprecation_warnings_as_errors is true)"`